genesis-tools/README.md

```diff
@@ -0,0 +1,31 @@
+# Genesis Tools
+
+* `genesis-populate` -- tool for creating genesis state dump populated with large number of accounts;
+* TODO `genesis-rebase`-- tool for rebasing the entire chain to a new genesis;
+* TODO `genesis-mainnet` -- tool for creating the main genesis used at the MainNet launch;
+
+## `genesis-populate`
+
+Performance of our node varies drastically depending on the size of the trie it operates with.
+As the baseline, we agreed to take the trie roughly equivalent to the trie of Ethereum (as of October 2019) in
+its complexity. We achieve it by populating trie with 80M accounts and uploading a smart contract to each of them.
+We also make sure the trie does not compress the subtrees due to similar account names. We then use this trie for
```

Comment:
Same question here. Also do we consider compress same contract deployed by different account?

Reply:
The compression happens when account names are such that they have lots of common prefixes and there are subtrees in trie that are identical if we disregard account name prefixes. This situation happens when account names are generated by just adding an integer to the name of the account `my_account_0`, `my_account_1`, etc. It seems clear to me that this situation is highly artificial and will not arise with real account names, and even if it ever does we should not be counting on it. In other words, we should be pessimistic when measuring performance of our system, rather than optimistic. Therefore I followed @mikhailOK 's advice to generate such account names that there is no compression.

Reply:
> we should be pessimistic when measuring performance of our system

Sounds good! Also answers "compress same contract deployed by different account"

Do we want compression in real system though? If we benchmark and prove compression can save a lot. It could happen when:
1. smart contract user create a lot of robot prefixed account
2. majority user doesn't write their own smart contract. majority of smart contract could be some most popular smart contract code.

Reply:
The real system uses compression.